framework/web/Request.php

```diff
@@ -1142,6 +1152,16 @@ public function getUserAgent()
      * @since 2.0.28
      */
     protected function getUserIpFromIpHeaders() {
+        if (($ip = $this->getForwardedHeaderPart('for')) !== null) {
```

Comment:
check only, if the `forwarded` header is contains in `$ipHeaders`.

Reply:
it will break `protected function getUserIpFromIpHeader($ips)`

because Forwarded header is more complicated than just list of IPs

Reply:
However, if you ignore `$ipHeaders`, anyone can inject the IP address by submitting the `Forwarded` header.

```http
GET / HTTP/1.1
x-forwarded-for: 8.8.8.8, 9.9.9.9
forwarded: for=127.0.0.1             <---- injected
```

Reply:
@samdark maybe if create a separate configuration parameter?
```php
[
  'class' => Request::class,
  'ipListRfc7239' => ['forwarded'],
]
```

Reply:
I could filter out `Forwarded` from `getUserIpFromIpHeader()` but that wouldn't be nice

Reply:
OR I could look every time in `$secureHeaders` array, that `Forwarded` is there
That way user can just say that `Forwarded` is not secure, by removing that from `$secureHeaders`

Reply:
> OR I could look every time in $secureHeaders array, that Forwarded is there

@samdark how about this?

Reply:
Yes. That's a good thing to do. It should be configured explicitly.

Reply:
Aaaand we should be fine